pkg/graveler/ref/manager.go

```diff
@@ -58,46 +66,58 @@ func protoFromBranch(branchID graveler.BranchID, b *graveler.Branch) *graveler.B
 }
 
 func NewKVRefManager(executor batch.Batcher, kvStore kv.StoreMessage, addressProvider ident.AddressProvider) *KVManager {
+	repoCache := cache.NewCache(RepositoryCacheSize, RepositoryCacheExpiry, cache.NewJitterFn(RepositoryCacheJitter))
 	return &KVManager{
 		kvStore:         kvStore,
 		addressProvider: addressProvider,
 		batchExecutor:   executor,
+		repoCache:       repoCache,
 	}
 }
 
 func (m *KVManager) getRepository(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.RepositoryRecord, error) {
+	data := graveler.RepositoryData{}
+	_, err := m.kvStore.GetMsg(ctx, graveler.RepositoriesPartition(), []byte(graveler.RepoPath(repositoryID)), &data)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrRepositoryNotFound
+		}
+		return nil, err
+	}
+	return graveler.RepoFromProto(&data), nil
+}
+
+func (m *KVManager) getRepositoryBatch(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.RepositoryRecord, error) {
 	key := fmt.Sprintf("GetRepository:%s", repositoryID)
 	repository, err := m.batchExecutor.BatchFor(ctx, key, MaxBatchDelay, batch.BatchFn(func() (interface{}, error) {
-		data := graveler.RepositoryData{}
-		_, err := m.kvStore.GetMsg(ctx, graveler.RepositoriesPartition(), []byte(graveler.RepoPath(repositoryID)), &data)
-		if err != nil {
-			return nil, err
-		}
-		return graveler.RepoFromProto(&data), nil
+		return m.getRepository(ctx, repositoryID)
 	}))
-	if errors.Is(err, kv.ErrNotFound) {
-		err = graveler.ErrRepositoryNotFound
-	}
 	if err != nil {
 		return nil, err
 	}
 	return repository.(*graveler.RepositoryRecord), nil
 }
 
 func (m *KVManager) GetRepository(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.RepositoryRecord, error) {
-	repo, err := m.getRepository(ctx, repositoryID)
+	rec, err := m.repoCache.GetOrSet(repositoryID, func() (interface{}, error) {
+		repo, err := m.getRepositoryBatch(ctx, repositoryID)
+		if err != nil {
+			return nil, err
+		}
+
+		switch repo.State {
+		case graveler.RepositoryState_ACTIVE:
+			return repo, nil
+		case graveler.RepositoryState_IN_DELETION:
+			return nil, graveler.ErrRepositoryInDeletion
+		default:
+			return nil, fmt.Errorf("invalid repository state (%d) rec: %w", repo.State, graveler.ErrInvalid)
```

Comment:
Is a number for a state indicative in the returned error?

Reply:
1. Just made the format explict - it was a number
2. When we will get this error it means we have a bug and we should handle the new repository state here or add a new error UnknownState and drop this format.
3. 
Let me know what you think I will handle it in a different PR.